packages/flutter_tools/test/integration.shard/android_plugin_compilesdkversion_mismatch_test.dart

```diff
@@ -0,0 +1,124 @@
+
+    // Create dummy plugin
+    processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'create',
+      '--template=plugin',
+      '--platforms=android',
+      'test_plugin',
+    ], workingDirectory: tempDir.path);
+
+    final Directory pluginAppDir = tempDir.childDirectory('test_plugin');
+    final File pluginGradleFile = pluginAppDir.childDirectory('android').childFile('build.gradle');
+    expect(pluginGradleFile, exists);
+    final String pluginBuildGradle = pluginGradleFile.readAsStringSync();
+
+
+    // Bump up plugin compileSdkVersion to 31
+    final String newPluginGradleFile = pluginBuildGradle.replaceAll(
+      androidCompileSdkVersionRegExp, 'compileSdkVersion 31');
+    pluginGradleFile.writeAsStringSync(newPluginGradleFile);
+
+    // Create dummy project
+    processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'create',
+      '--platforms=android',
+      'test_project',
+    ], workingDirectory: tempDir.path);
+
+    final Directory projectAppDir = tempDir.childDirectory('test_project');
+    final File pubspecFile = projectAppDir.childFile('pubspec.yaml');
+    expect(pubspecFile, exists);
+    final File projectGradleFile = projectAppDir.childDirectory('android').childDirectory('app').childFile('build.gradle');
+    expect(projectGradleFile, exists);
+    final String projectBuildGradle = projectGradleFile.readAsStringSync();
+    final String pubspecFileString = pubspecFile.readAsStringSync();
+
+    // Bump down project compileSdkVersion to 30
+    final String newProjectGradleFile = projectBuildGradle.replaceAll(
+      androidCompileSdkVersionRegExp, 'compileSdkVersion 30');
+    projectGradleFile.writeAsStringSync(newProjectGradleFile);
+
+    // Add dummy plugin as dependency to dummy project
+    final String newPubspecFile= pubspecFileString.replaceFirst(
+      'dependencies:', 'dependencies:\n  test_plugin:\n    path: ../test_plugin');
+    pubspecFile.writeAsStringSync(newPubspecFile);
+
+    // Run flutter pub get to update the dependencies
+    final BufferLogger logger = BufferLogger.test();
+    final Pub pub = Pub(
+        fileSystem: fileSystem,
+        logger: logger,
+        processManager: processManager,
+        platform: FakePlatform(),
+        botDetector: const FakeBotDetector(false),
+        usage: TestUsage(),
+      );
+
+    await pub.get(
+      context: PubContext.flutterTests,
+      directory: projectAppDir.path);
+
+    // Run flutter build apk to build dummy project
+    final ProcessResult result = processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'build',
+      'apk',
+      '--target-platform=android-arm',
+    ], workingDirectory: projectAppDir.path);
+
+    // Check error message is thrown
+    final RegExp charactersToIgnore = RegExp(r'\|/|[\n]');
```

Comment:
nit: these characaters are also valid in the test.  It could use Dart multiline strings if that is easier. e.g.

```dart
contains('''One or more plugins require a higher Android SDK version.

Fix this issue by adding the following to ${projectGradleFile.path}:

android {
   compileSdkVersion 31
}
''')

Reply:
@blasten There was one test that added a newline and changed one '\\' to a '/' and it was failing on that. Can you think of an alternate solution?

Reply:
@blasten There is one test that changes a '\\' to a '/' for some weird reason. Can you suggest an alternate solution to ignoring the slashes?